molecule/provisioners/libvirtprovisioner.py

```diff
@@ -0,0 +1,923 @@
+                    os.path.join(self._pool_path, image['name'] + '.img')):
+                if not os.path.exists(
+                        os.path.join(self._sources_path, imagefile)):
+                    self._fetch(image['source'], imagefile)
+                if imagefile.endswith('.box'):
+                    self._unpack_box(image, imagefile)
+
+            # Is there an existing libvirt volume for this instance?
+            try:
+                vol_found = pool.storageVolLookupByName(instance['name'])
+                if vol_found and not vol_found.isActive():
+                    vol_found.create()
+                return vol_found  # existing volume is now running
+            except libvirt.libvirtError as e:
+                self._create_volume(pool, instance)
+            # TODO: Are all this instance's interface's networks available?
+            # Is there an existing libvirt domain defined for this instance?
+            dom_found = False
+            for dom in domains:
+                if not dom_found:
+                    if dom.name() == instance['name']:
+                        dom_found = True
+                        if dom.info()[0] == 1:
+                            utilities.print_info(
+                                "\t{}: already running".format(instance[
+                                    'name']))
+                            break
+                        else:
+                            utilities.print_info("\t{}: booting".format(
+                                instance['name']))
+                            dom.create()
+                            if wait:
+                                time.sleep(self._boot_wait)
+                                wait = False
+            if not dom_found:
+                utilities.print_info("\t{}: defining".format(instance['name']))
+                dom = self._libvirt.defineXML(self._build_domain_xml(instance))
+                if GUESTFS:
+                    # TODO: Here is the point at which to modify with libguestfs,
+                    #  i.e. after creating the domain's volume but before booting the domain.
+                    # if any of this instance's interfaces has a key 'address'? or always?
+                    try:
+                        guest = guestfs.GuestFS(python_return_dict=True)
+                        self._manipulate_guest_image(guest, instance)
+                        guest.umount_all()
+                        guest.shutdown()
+                        guest.close()
+                    except Exception, e:
+                        LOG.warning("\nFAILED to manipulate the guest image so could not configure network interfaces: {}".format(e))
+                utilities.print_success("\tCreated instance {}.\n".format(
+                    instance['name']))
+                utilities.print_info("\t{}: booting".format(instance['name']))
+                try:
+                    dom.create()
+                    if wait:
```

Comment:
This seems fishy, if we haven't waited before we just wait a little here and hope for the best?
For this (and the similar wait above) I suggest adding a tiny utility method that could look like this:
```
def wait_until(somepredicate, timeout, period=0.25, *args, **kwargs):
  mustend = time.time() + timeout
  while time.time() < mustend:
    if somepredicate(*args, **kwargs): return True
    time.sleep(period)
  return False
```